superset/reports/commands/delete.py

```diff
@@ -0,0 +1,55 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import logging
+from typing import Optional
+
+from flask_appbuilder.models.sqla import Model
+from flask_appbuilder.security.sqla.models import User
+
+from superset.commands.base import BaseCommand
+from superset.dao.exceptions import DAODeleteFailedError
+from superset.models.reports import ReportSchedule
+from superset.reports.commands.exceptions import (
+    ReportScheduleDeleteFailedError,
+    ReportScheduleNotFoundError,
+)
+from superset.reports.dao import ReportScheduleDAO
+
+logger = logging.getLogger(__name__)
+
+
+class DeleteReportScheduleCommand(BaseCommand):
+    def __init__(self, user: User, model_id: int):
+        self._actor = user
+        self._model_id = model_id
+        self._model: Optional[ReportSchedule] = None
+
+    def run(self) -> Model:
+        self.validate()
+        try:
+            report_schedule = ReportScheduleDAO.delete(self._model)
+        except DAODeleteFailedError as ex:
+            logger.exception(ex.exception)
+            raise ReportScheduleDeleteFailedError()
+        return report_schedule
+
+    def validate(self) -> None:
+        # Validate/populate model exists
+        self._model = ReportScheduleDAO.find_by_id(self._model_id)
+        if not self._model:
+            raise ReportScheduleNotFoundError()
+        # TODO check integrity
```

Comment:
Let's not leave TODOs lying around. Can we address in this PR?

Reply:
It's already addressed, this is a left over, like a note to self. (deleted)